pkg/kubelet/envvars/envvars.go

```diff
@@ -60,15 +66,19 @@ func FromServices(services *api.ServiceList) []api.EnvVar {
 }
 
 func makeEnvVariableName(str string) string {
-	// TODO: If we simplify to "all names are DNS1123Subdomains" this
-	// will need two tweaks:
-	//   1) Handle leading digits
-	//   2) Handle dots
-	return strings.ToUpper(strings.Replace(str, "-", "_", -1))
+	return strings.ToUpper(strings.Replace(strings.Replace(str, ".", "_", -1), "-", "_", -1))
```

Comment:
@bacongobbler toleration of '.' and number as the first char...do they belong in another PR? (since it's not mentioned in the commit message?